### ISSUE TITLE
LKB Configurator Not Recognising Keyboard

### ISSUE THREAD

**Issue by pseudobacon** [Lotlab/Lot60-BLE-Keyboard#3]
I have built a Lot60-BLE Rev.F and also a GT-BLE60 and neither keyboard will be recognised by LKB Configurator. I have also bought a Lot60-BLE Rev.F that was assembled from the Taobao store. This keyboard is recognised by LKB Configurator.

I didn't get any information from LKB Configurator on [Glab](http://wiki.glab.online/#!download.md) so I downloaded version 0.9.9.10 which requires a web browser but gives a command prompt output. For the PCB purchased from Taobao, there are no messages shown. But for the Lot60-BLE and GT-BLE60 that I built I get the following messages:

![image](https://user-images.githubusercontent.com/12183854/95679892-72898000-0c21-11eb-9e1b-aed8ef5144a1.png)

I tried compiling the CH554 bootloader myself, and also used the bootloader binaries available here for the BLE60, and the bootloader binary available for the GT-BLE60 on Glab. I used WCHISPTool V2.80 to flash the bootloader.

I noticed that when my PCBs were plugged in, the USB PID and VID would be different to the Taobao PCB.

Taobao:
![image](https://user-images.githubusercontent.com/12183854/95679994-12dfa480-0c22-11eb-9cfd-22e0b2952363.png)

Self assembled:
![image](https://user-images.githubusercontent.com/12183854/95680167-4242e100-0c23-11eb-9c08-1ec92bf748fa.png)

The VID and PID is set in the [config file](keyboard/lot60-ble/common/config.h) but nowhere can I see that Col is set, so I don't understand why my two self assembled keyboards have this set.

The source code for LKB Configurator is not available, so my guess is that the configurator looks for a particular PID and VID value. This value does not change whether or not I build the bootloader from source, or when I flash the precompiled binaries. So perhaps I have faulty CH552 and CH554 chips? (I have tried both. I purchased the CH552 from [优信](https://item.taobao.com/item.htm?id=574979089030) so I don't feel like they are fake. I also tried multiple chips with the same result.

Finally, when I use the LKB Configurator to generate the EEP data, it will also not flash with KeymapDownloader and just says Operation timed out:
![image](https://user-images.githubusercontent.com/12183854/95680134-10317f00-0c23-11eb-8f8d-d93b501b7fd9.png)

**Comment by jim-kirisame** [comment]
Hello, the firmware of Lot60-BLE is directly compile from the source and has no modification. There is also no any limitation in USB firmware. The reason why the configurator doesn't recoginze is the hid operation timeout. Old version firmware (before 1.1.0.0) does not support configurate via the configurator, and it will show timeout. If you are using the latest precompile firmware, that might be the problem of hardware. Try to connect your device directly to your computer (without usb hub).

**Comment by pseudobacon** [comment]
I tried a different computer but still the same result. Here is a video of the flashing process for the bootloader that I did:
https://streamable.com/tlo1ut

It seems strange that the bootloader will flash, but when writing to EEPROM there is a timeout. It seems unlikely that all of my CH552 and CH554 chips would be faulty (but its possible). I did an inspection of the Taobao board and the only other differences that I could see were very minor:

- XC6206P332MR Taobao - chip is marked 6628, and is physically bigger and more rectangular. Mine is marked 662K

- TP4057 Taobao - chip is marked 57ba. Mine is marked 57b6

- B5819W Taobao - I used B5817W here (probably should change to at least B5818W since Type-C is 20V)

I assume all other capacitors and resistors are correct, otherwise the bootloader should not have written successfully. Any ideas on what component could be faulty here? I have tested more than one PCB, including the GT-BLE60 which was printed by a different manufacturer.

**Comment by jim-kirisame** [comment]
Hello, is your device work correctly? And which version of firmware you are using? 

Using the configurator in the https://github.com/Lotlab/lkb-configurator/releases , the configurator you use in the video will not output any debug log.

**Comment by pseudobacon** [comment]
It doesn't work. There is no keymap when the bootloader is flashed, so the PCB is useless unless you can successfully flash the EEP file to the board. 

I used firmware version 1.1.3.2. I uploaded the video to bilibili instead: https://www.bilibili.com/video/BV1ji4y1j7Fj/

In this video I used the configurator from Glab http://wiki.glab.online/#!download.md
![image](https://user-images.githubusercontent.com/12183854/95797546-81645580-0d3b-11eb-970e-43f955757989.png)

When I tried the configurator from https://github.com/Lotlab/lkb-configurator/releases, I got the System.TimeoutException: Operation timed out error

**Comment by jim-kirisame** [comment]
There is a default configuration in the firmware, so if your keyboard doesn't work, please check your bluetooth firmware.

BTW, the bluetooth firmware is depending on the Softdevice, so you have to flash that on the new chip.

**Comment by pseudobacon** [comment]
Ok. I got the GT-BLE60 board working. I flashed a nrf52_kbd.hex that I compiled from source with pyocd, which then prompted the keyboard to boot into bootloader mode with DfuTarg. I then uploaded the bluetooth firmware zip using nRF Connect. Now that keyboard is recognised by LKB-Configurator 👍.

![image](https://user-images.githubusercontent.com/12183854/95811436-06149b00-0d5f-11eb-81e5-3a22dcb091c4.png)

So I guess this means that application is not flashed correctly on my Lot-BLE60. I believe the softdevice is okay, since the keyboard connects over bluetooth. Its just that the keys do not work. 

Unfortunately right now the keyboard is not responding to my Raspberry Pi using openocd, and also it is not responding to my clone J-Link OB. I will build a DAPLink and see if I can recover the NRF chip with that I guess.
